### ISSUE TITLE
Support for subnetwork on google_container_cluster resource

### ISSUE THREAD

**Issue by devth** [hashicorp/terraform#5551]
https://cloud.google.com/sdk/gcloud/reference/container/clusters/create

**Comment by devth** [comment]
Currently if a Network is in custom mode, it's impossible to create a cluster since Subnetwork is a required parameter in that case:

```
* google_container_cluster.standard-test-public: googleapi: Error 400: Subnetwork must be provided for manual-subnet Network "test-m"., badRequest
```

**Comment by lwander** [comment]
We try to avoid adding support for non-GA features, a good explanation is given here: https://github.com/hashicorp/terraform/issues/5053#issuecomment-182096731.

**Comment by devth** [comment]
The `self_link` issue may not apply here. For example, in #5552 I was able to work around by specifying a string name instead of relying on `self_link`.

**Comment by lwander** [comment]
The issue is that if anyone has an existing resource that refers to a cluster with something like `${google_container_cluster.name.self_link}`, that reference's value will change and include `alpha`. The resource referring to the cluster may then have to be destroyed and recreated, which is undesirable.

**Comment by JorritSalverda** [comment]
I don't think subnetwork is in beta, seeing it's part of stable gcloud:

https://cloud.google.com/sdk/gcloud/reference/container/clusters/create

I get the following error when trying to create a google_container_cluster in a project with manually configured subnets:

`* google_container_cluster.staging: googleapi: Error 400: Subnetwork must be provided for manual-subnet Network "staging"., badRequest
`

**Comment by JorritSalverda** [comment]
Sorry, a closer look shows me it's an alpha feature even though it's in stable gcloud command.

**Comment by devth** [comment]
Fixed by #5871
